docs/targets/solana.rst

```diff
@@ -227,16 +227,20 @@ is passed to the transaction that runs the constructor code.
 
 Alternatively, the data account can be created by the constructor, on chain. When
 this method is used, some parameters must be specified for the account
-using annotations. Those are placed before the constructor. If there is no
-constructor present, then an empty constructor can be added. The constructor
-arguments can be used in the annotations.
+using annotations. Annotations placed on top of a constructor can only contain literals or
```

Comment:
How about replacing "placed on top of" with above or before

"on top of" sounds like it needs to be on the same line somehow